static/app/views/settings/account/accountIdentities.tsx

```diff
@@ -34,54 +43,146 @@ class AccountIdentities extends AsyncView<Props, State> {
     return t('Identities');
   }
 
-  handleDisconnect = (identity: Identity) => {
-    const {identities} = this.state;
-
-    this.setState(
-      state => {
-        const newIdentities = state.identities?.filter(({id}) => id !== identity.id);
-
-        return {
-          identities: newIdentities ?? [],
-        };
-      },
-      () =>
-        disconnectIdentity(identity).catch(() => {
-          this.setState({
-            identities,
-          });
-        })
+  renderItem = (identity: UserIdentityConfig) => {
+    return (
+      <IdentityPanelItem key={`${identity.category}:${identity.id}`}>
+        <InternalContainer>
+          <PluginIcon pluginId={identity.provider.key} size={36} />
+          <IdentityText isSingleLine={!identity.dateAdded}>
+            <IdentityName>{identity.provider.name}</IdentityName>
+            {identity.dateAdded && <IdentityDateTime date={moment(identity.dateAdded)} />}
+          </IdentityText>
+        </InternalContainer>
+        <InternalContainer>
+          <TagWrapper>
+            <Tag
+              isOrgName={false}
+              label={
+                identity.category === UserIdentityCategory.SOCIAL_IDENTITY
+                  ? t('App Integration')
+                  : identity.category === UserIdentityCategory.GLOBAL_IDENTITY
+                  ? t('Sign-In Identity')
+                  : identity.category === UserIdentityCategory.ORG_IDENTITY
+                  ? t('Organization Identity')
+                  : ''
+              }
+            />
+            {identity.organization && (
+              <Tag isOrgName label={identity.organization.slug} />
+            )}
+          </TagWrapper>
+
+          {this.renderButton(identity)}
+        </InternalContainer>
+      </IdentityPanelItem>
     );
   };
 
+  renderButton(identity: UserIdentityConfig) {
+    return identity.status === UserIdentityStatus.CAN_DISCONNECT ? (
+      <Confirm
+        onConfirm={() => this.handleDisconnect(identity)}
+        priority="danger"
+        confirmText={t('Disconnect')}
+        message={
+          <React.Fragment>
+            <Alert type="error" icon={<IconFlag size="md" />}>
+              {t(`Disconnect Your ${identity.provider.name} Identity?`)}
+            </Alert>
+            <TextBlock>
+              {identity.category === UserIdentityCategory.SOCIAL_IDENTITY
+                ? t("This action can't be undone.")
+                : t(
+                    'After disconnecting, you will need to use a password or another identity to sign in.'
+                  )}
+            </TextBlock>
+          </React.Fragment>
+        }
+      >
+        <Button>{t('Disconnect')}</Button>
```

Comment:
Let's go-ahead and make this button `size="small"`

Reply:
```suggestion
        <Button size="small">{t('Disconnect')}</Button>
```

Reply:
Looks okay to me. I'll run it by our design stakeholders.

Reply:
![image](https://user-images.githubusercontent.com/1421724/140235643-e2dc1097-7930-4531-ac42-54082ff168f5.png)

the rows are def too big haha

Reply:
Sorry, that screenshot is out of date. Here is a comparison with and without `size="small"`

<img width="1440" alt="Screen Shot 2021-11-03 at 5 07 45 PM" src="https://user-images.githubusercontent.com/26411900/140236000-7063cfb3-bb0d-4051-806f-4964182ba218.png">
<img width="1440" alt="Screen Shot 2021-11-03 at 5 07 51 PM" src="https://user-images.githubusercontent.com/26411900/140236003-27ebef39-4488-4415-b641-a899337b1c90.png">

Reply:
Let's use `small`. Seems to look more balanced here.